galite-core/src/main/kotlin/org/kopi/galite/visual/ApplicationContext.kt

```diff
@@ -15,18 +15,329 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
-
 package org.kopi.galite.visual
 
+import org.kopi.galite.base.UComponent
 import org.kopi.galite.db.DBContext
+import org.kopi.galite.base.Utils
+import org.kopi.galite.l10n.LocalizationManager
+import org.kopi.galite.util.mailer.Mailer
+
+import java.io.CharArrayWriter
+import java.io.StringWriter
+import java.io.PrintWriter
+import java.io.InputStreamReader
+import java.io.FileWriter
+import java.io.IOException
+import java.net.NetworkInterface
+import java.net.SocketException
+import java.util.Locale
+import java.util.Date
 
+/**
+ * `ApplicationContext` is a galite application context that contains the
+ * running [Application] instance in the context thread. The `ApplicationContext`
+ * handles all shared applications components.
+ */
 abstract class ApplicationContext {
-  abstract val application: Application
+  //-----------------------------------------------------------
+  // ABSTRACT METHODS
+  //-----------------------------------------------------------
+  /**
+   * Returns the **current** [Application] instance.
+   * @return The **current** [Application] instance.
+   */
+  abstract fun getApplication(): Application
+
+  /**
+   * Returns the **current** [PreviewRunner] instance.
+   * @return The **current** [PreviewRunner] instance.
+   */
+  abstract fun getPreviewRunner(): PreviewRunner
+
+  /**
+   * Returns `true` if we are in a web application context.
+   * @return `true` if we are in a web application context.
+   */
+  abstract fun isWebApplicationContext(): Boolean
+
   companion object {
-    fun getDBContext(): DBContext {
-      TODO()
+
+    //-----------------------------------------------------------
+    // UTILS
+    //-----------------------------------------------------------
+
+    /**
+     * Returns the default configuration of the Application
+     */
+    fun getDefaults(): ApplicationConfiguration? = applicationContext.getApplication().getApplicationConfiguration()
```

Comment:
fun getDefaults(): ApplicationConfiguration = applicationContext.getApplication().getApplicationConfiguration()

and please fix all null checks for ApplicationConfiguration and all other variables